cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/dataset/MetadataDatasetTest.java

```diff
@@ -1137,42 +1137,34 @@ public void apply() throws Exception {
         searchResults = dataset.search(namespaceId, "*", targets, nameAsc, 0, 2, 0, null, false,
                                        EnumSet.allOf(EntityScope.class));
         Assert.assertEquals(ImmutableList.of(flowEntry, dsEntry), searchResults.getResults());
-        Assert.assertEquals(ImmutableList.of(flowEntry, dsEntry, appEntry), searchResults.getAllResults());
```

Comment:
`getAllResults()` should get all search results, including the ones in the offset. This way, we'll be able to count the total.